### PR TITLE
CR-1077626: graph status thread cleanup

### PR DESCRIPTION
**Pull request by Himanshu-xilinx**
fix below crash
"
terminate called without an active exception
Aborted
"

### CODE REVIEW

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c [hunk 1]
```diff
@@ -272,8 +272,12 @@ zocl_aie_getcmd_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
 	mutex_lock(&aie->aie_lock);
 	while (list_empty(&aie->aie_cmd_list)) {
 		mutex_unlock(&aie->aie_lock);
-		if (wait_event_interruptible(aie->aie_wait_queue,
-		    !list_empty(&aie->aie_cmd_list))) {
+		/* return greater then 0 if contition true before timeout,
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_aie.c:275]: typo?

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c [hunk 2]
```diff
@@ -272,8 +272,12 @@ zocl_aie_getcmd_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
...
+ 		 * 0 when time out, else -ERESTARTSYS.
+ 		 */
+		int ret = wait_event_interruptible_timeout (aie->aie_wait_queue,
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_aie.c:278]: the compiler might not be happy by defining a variable at middle of a scope

**Review by larry9523 — COMMENTED** [review]
Some minor comments

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c (hunk 2 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_aie.c]: done

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c (hunk 1 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_aie.c]: fixed

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c [hunk 3]
```diff
@@ -272,8 +273,13 @@ zocl_aie_getcmd_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
 	mutex_lock(&aie->aie_lock);
 	while (list_empty(&aie->aie_cmd_list)) {
 		mutex_unlock(&aie->aie_lock);
-		if (wait_event_interruptible(aie->aie_wait_queue,
-		    !list_empty(&aie->aie_cmd_list))) {
+		/* 
+ 		 * Return greater then 0 if condition true before timeout,
+ 		 * 0 when time out, else -ERESTARTSYS.
+ 		 */
+		ret = wait_event_interruptible_timeout (aie->aie_wait_queue,
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_aie.c:280]: Please remove the blank space between function name and '('

**Review by larry9523 — APPROVED** [review]
Please remove the blank space

File: src/runtime_src/core/edge/user/aie/aied.cpp [hunk 4]
```diff
@@ -44,6 +51,9 @@ Aied::pollAIE()
   while (1) {
     /* Calling XRT interface to wait for commands */
     if (drv->xclAIEGetCmd(&cmd) != 0) {
+      /* break if destructor called */
+      if(done == true)
```
Comment (keryell) [inline src/runtime_src/core/edge/user/aie/aied.cpp:55]:
```suggestion
      if (done)
```

**Review by keryell — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c [hunk 5]
```diff
@@ -272,8 +273,13 @@ zocl_aie_getcmd_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
 	mutex_lock(&aie->aie_lock);
 	while (list_empty(&aie->aie_cmd_list)) {
 		mutex_unlock(&aie->aie_lock);
-		if (wait_event_interruptible(aie->aie_wait_queue,
-		    !list_empty(&aie->aie_cmd_list))) {
+		/* 
+ 		 * Return greater then 0 if condition true before timeout,
```
Comment (larry9523) [inline src/runtime_src/core/edge/drm/zocl/zocl_aie.c:277]: than?

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c (hunk 3 above)
Comment (Himanshu-xilinx) [reply src/runtime_src/core/edge/drm/zocl/zocl_aie.c]: fixed

**Review by chvamshi-xilinx — APPROVED** [review]

**Pull request merged**